subprojects/docs/src/docs/userguide/javaTesting.adoc

```diff
@@ -15,8 +15,55 @@
 
+It explains:
+
+ * Ways to control how the tests are run (<<sec:test_execution,Test execution>>)
+ * How to select specific tests to run (<<sec:test_filtering,Test filtering>>)
+ * What test reports are generated and how to influence the process (<<test_reporting,Test reporting>>)
+ * How Gradle finds tests to run (<<sec:test_detection,Test detection>>)
+ * How to make use of the major frameworks' mechanisms for grouping tests together (<<test_grouping,Test grouping>>)
+
+But first, we look at the basics of JVM testing in Gradle.
+
+[[sec:java_testing_basics]]
+=== The basics
+
+All JVM testing revolves around a single task type: api:org.gradle.api.tasks.testing.Test[]. This runs a collection of test cases using any supported test library — JUnit, JUnit Platform or TestNG — and collates the results. You can then turn those results into a report via an instance of the api:org.gradle.api.tasks.testing.TestReport[] task type.
+
+In order to operate, the `Test` task type requires just two pieces of information:
+
+ * Where to find the compiled test classes (property: api:org.gradle.api.tasks.testing.Test#getTestClassesDirs()[])
+ * The execution classpath, which should include the classes under test as well as the test library that you're using (property: api:org.gradle.api.tasks.testing.Test#getClasspath()[])
+
+When you're using a JVM language plugin — such as the <<java_plugin,Java Plugin>> — you will automatically get the following:
+
+ * A dedicated `test` source set for unit tests
+ * A `test` task of type `Test` that runs those unit tests
+ 
+The JVM language plugins use the source set to configure the task with the appropriate execution classpath and the directory containing the compiled test classes. In addition, they attach the `test` task to the `check` <<sec:lifecycle_tasks,lifecycle task>>.
+
+It's also worth bearing in mind that the `test` source set automatically creates <<java_source_set_configurations,corresponding dependency configurations>> — of which the most useful are `testImplementation` and `testRuntimeOnly` — that the plugins tie into the `test` task's classpath.
+
+All you need to do in most cases is configure the appropriate compilation and runtime dependencies and add any necessary configuration to the `test` task. The following example shows a simple setup that uses JUnit 4.x and changes the maximum heap size for the tests' JVM to 1 gigabyte:
+
+++++
+<sample id="basicJavaTestConfig" dir="userguide/java/basic" title="A basic configuration for the 'test' task">
+    <sourcefile file="build.gradle" snippet="java-basic-test-config"/>
+</sample>
+++++
+
+The `Test` task has many configuration options and we cover a significant number of them in the rest of the chapter.
```

Comment:
❓ Do you think some kind of reference to the DSL reference would be appropriate around here?

Reply:
It's a good question and one I ask myself. Since there is already a link to `Test` at the beginning of that section, I lean towards not including a link. If it were in a different section, I would be more inclined to make it a link. That said, the reference to many configuration options looks like a good candidate for a link.